### DISCUSSION TITLE
Auth: Save cookies and Redirect

### DISCUSSION THREAD

**Discussion by geam-dev** [Chaoyingz/flect#11] [Q&A]
Hi! I'm building an internal application for the company I work for. There are certain resources that need to be kept private, and I'm having trouble redirecting while also storing the token in the browser's cookies.

How should I do this? With FastAPI, I would normally use a RedirectResponse(url="", status_code=302, headers={"set-cookie": "key:value"}). However, here, I can't pass these parameters to the ActionResponse(action=Redirect(path="")).

Thank you very much!

(Message translated by IA, My English is not very good)

**Comment by Chaoyingz** [comment]
Sure, you can use FastAPI's `RedirectResponse` directly, like this:

```python
from fastapi import Request, Response
from starlette.responses import RedirectResponse

async def get(
    request: Request,
    response: Response,
) -> RedirectResponse:
    response.set_cookie(key="fakesession", value="fake-cookie-session-value")
    return RedirectResponse("/")
```

This way, you can set the cookie and perform the redirection in one go.

**Comment by geam-dev** [comment]
Sorry for the question, haha, today I tried a bit more lucidly and my mistake was in the redirection link I was using. Today I was able to with the Redirect itself, I'll copy the code in case it helps someone else, thank you very much.

```
from fastapi import Response
from flect import ActionResponse
from flect.actions import Redirect
from src.e_intranet.app.models.forms import LoginForm

async def post(
    form: LoginForm,
    response: Response,
    ) -> ActionResponse:
    response.set_cookie(key="fakesession", value="fake-cookie-session-value")
    return ActionResponse(
        action=Redirect(
            path="/dashboard"
        )
    )
```
